pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -359,6 +412,397 @@ func (r *Reconciler) reconcileDeployment(ctx context.Context, logger *zap.Sugare
+		Name:        customObjectData.Spec.Template.Name,
+		Labels:      customObjectData.Spec.Template.Labels,
+		Annotations: customObjectData.Spec.Template.Annotations,
+	}
+	original.Spec.Template.Spec = corev1.PodSpec{
+		Tolerations:        customObjectData.Spec.Template.Spec.Tolerations,
+		NodeSelector:       customObjectData.Spec.Template.Spec.NodeSelector,
+		ServiceAccountName: customObjectData.Spec.Template.Spec.ServiceAccountName,
+		Containers:         []corev1.Container{container},
+		Volumes: []corev1.Volume{{
+			Name: "config-logging",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: eventListenerConfigMapName,
+					},
+				},
+			},
+		}},
+	}
+	marshaledData, err := json.Marshal(original)
+	if err != nil {
+		logger.Errorf("failed to marshal custom object", err)
+		return err
+	}
+	data := new(unstructured.Unstructured)
+	if err := data.UnmarshalJSON(marshaledData); err != nil {
+		logger.Errorf("failed to unmarshal to unstructured object", err)
+		return err
+	}
+
+	if data.GetName() == "" {
+		data.SetName(el.Status.Configuration.GeneratedResourceName)
+	}
+	gvr, _ := meta.UnsafeGuessKindToResource(data.GetObjectKind().GroupVersionKind())
+
+	data.SetOwnerReferences([]metav1.OwnerReference{*el.GetOwnerReference()})
+
+	var watchError error
+	r.onlyOnce.Do(func() {
+		watchError = r.podspecableTracker.WatchOnDynamicObject(ctx, gvr)
+	})
+	if watchError != nil {
+		logger.Errorf("failed to watch on created custom object", watchError)
+		return err
+	}
+
+	existingCustomObject, err := r.DynamicClientSet.Resource(gvr).Namespace(namespace).Get(ctx, data.GetName(), metav1.GetOptions{})
+	switch {
+	case err == nil:
+		if reconcileCustomObject(existingCustomObject, data) {
+			if _, err := r.DynamicClientSet.Resource(gvr).Namespace(namespace).Update(ctx, existingCustomObject, metav1.UpdateOptions{}); err != nil {
+				logger.Errorf("Error updating to eventListener custom object: %v", err)
+				return err
+			}
+			logger.Infof("Updated EventListener Custom Object %s in Namespace %s", data.GetName(), el.Namespace)
+		}
+
+		statusData, ok, err := unstructured.NestedMap(existingCustomObject.Object, "status")
+		if !ok {
+			// No status in the created object, it is weird but let's not fail
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		conditionData, ok, err := unstructured.NestedFieldCopy(statusData, "conditions")
+		if !ok {
+			// No conditions in the created object, it is weird but let's not fail
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		generationData, ok, err := unstructured.NestedInt64(statusData, "observedGeneration")
+		if !ok {
+			// No observedGeneration in the created object, it is weird but let's not fail
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		// set generation value to el status
+		el.Status.ObservedGeneration = generationData
```

Comment:
hmmm why do we need to do this?

Reply:
(we don't update observedGeneration from k8s deployment to EL deployment)

Reply:
ya its not needed removed :+1: